onionbalance/hs_v3/service.py

```diff
@@ -294,67 +316,102 @@ def _publish_descriptor(self, is_first_desc):
         """
         from onionbalance.hs_v3.onionbalance import my_onionbalance
 
-        if not self._should_publish_descriptor_now(is_first_desc):
-            logger.info("No reason to publish %s descriptor for %s",
-                        "first" if is_first_desc else "second",
-                        self.onion_address)
-            return
-
-        try:
-            intro_points = self._get_intros_for_desc()
-        except NotEnoughIntros:
-            return
+        max_intro_points = params.MAX_INTRO_POINTS
 
-        # Derive blinding parameter
-        _, time_period_number = hashring.get_srv_and_time_period(is_first_desc)
-        blinding_param = my_onionbalance.consensus.get_blinding_param(self._get_identity_pubkey_bytes(),
-                                                                      time_period_number)
+        distinct_descriptors = params.DISTINCT_DESCRIPTORS
```

Comment:
I like the idea of using distinct descriptors only when it's needed, so I really like the logic here. In this case, let's not make distinct descriptors be a global parameter, and let's just have this logic here to decide whether to enable this feature or not.

We don't need to expose this "distinct descriptor" feature to the user, and onionbalance can just intelligently decide whether to enable it or not. I like this.